app/scripts/modules/cloudfoundry/src/pipeline/stages/deployService/cloudfoundryDeployServiceStage.module.ts

```diff
@@ -0,0 +1,40 @@
+import { IController, IScope, module } from 'angular';
+import { react2angular } from 'react2angular';
+
+import { CloudfoundryDeployServiceStageConfig } from './CloudfoundryDeployServiceStageConfig';
+import { ExecutionDetailsTasks, IStage, Registry } from '@spinnaker/core';
+import { CloudfoundryDeployServiceExecutionDetails } from 'cloudfoundry/pipeline/stages/deployService/CloudfoundryDeployServiceExecutionDetails';
+
+class CloudFoundryDeployServiceStageCtrl implements IController {
+  constructor(public $scope: IScope) {
+    'ngInject';
+  }
+}
+
+export const CLOUD_FOUNDRY_DEPLOY_SERVICE_STAGE = 'spinnaker.cloudfoundry.pipeline.stage.deployServiceStage';
+module(CLOUD_FOUNDRY_DEPLOY_SERVICE_STAGE, [])
+  .config(function() {
+    Registry.pipeline.registerStage({
+      accountExtractor: (stage: IStage) => stage.context.credentials,
+      configAccountExtractor: (stage: IStage) => [stage.credentials],
+      provides: 'deployService',
+      key: 'deployService',
+      cloudProvider: 'cloudfoundry',
+      templateUrl: require('./cloudfoundryDeployServiceStage.html'),
```

Comment:
You can register a react component using `component: CloudfoundryDeployServiceStageConfig` instead of the `templateUrl` and `controller`.  Then you can delete the `react2angular` wrapper altogether.

Reply:
The suggested change could only be applied to `cloudfoundryDeployServiceStage.module.ts` and `cloudfoundryDestroyServiceStage.module.ts` because there was still a reliance on values (IE `application`) to be auto-injected.

Reply:
hmm, interesting.  I will take a look.

Reply:
It looks like the _only_ injected value is `application`?  That value is provided to the stage config react component as a prop.  So I suspect you could get the other `.module.ts` files the same treatment.

See here:

https://github.com/spinnaker/deck/blob/89977dc1323f3bb658ae6221b1b28665dd94d4d6/app/scripts/modules/core/src/pipeline/config/stages/stage.module.js#L194-L204

Reply:
We also make use of pipeline which is not set in the mentioned code:

https://github.com/spinnaker/deck/blob/89977dc1323f3bb658ae6221b1b28665dd94d4d6/app/scripts/modules/core/src/pipeline/config/stages/stage.module.js#L194-L204

we also noticed that not using shim causes other problems if the stage is grouped with other providers. For example if you have multiple providers for a particular stage and not all providers are written in React, then the ones in React will not be rendered because there is still code that requires all the stages to be Angular.

here:
https://github.com/spinnaker/deck/blob/89977dc1323f3bb658ae6221b1b28665dd94d4d6/app/scripts/modules/core/src/pipeline/config/stages/stage.module.js#L186-L189

The `BaseProviderStageCtrl` uses templateUrl to switch between providers as seen here:

https://github.com/spinnaker/deck/blob/0525282912db2987ea8167eb4749e14530c47da9/app/scripts/modules/core/src/pipeline/config/stages/baseProviderStage/baseProviderStage.js#L53-L59

If we remove the wrapper and only use React we would have larger refactor of how `BaseProviderStageCtrl`is implemented.

Reply:
I see, thanks for the explanation

Reply:
@alanmquach PTAL -- we should address these issues so CF can use react only stages

Reply:
Can we merge this and as a separate PR we will remove the wrapper?